packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -698,36 +694,28 @@ export abstract class FluidDataStoreContext
 
 	/**
 	 * After GC has run, called to notify the data store of routes used in it. These are used for the following:
-	 *
 	 * 1. To identify if this data store is being referenced in the document or not.
-	 *
 	 * 2. To determine if it needs to re-summarize in case used routes changed since last summary.
-	 *
-	 * 3. These are added to the summary generated by the data store.
-	 *
-	 * 4. To notify child contexts of their used routes. This is done immediately if the data store is loaded.
-	 * Else, it is done when realizing the data store.
-	 *
-	 * 5. To update the timestamp when this data store or any children are marked as unreferenced.
+	 * 3. To notify child contexts of their used routes. This is done immediately if the data store is loaded.
+	 * Else, it is done by the data stores's summarizer node when child summarizer nodes are created.
```

Comment:
Why not do the propagation via summarizer node always?  i.e. this function is 1-liner that calls `this.summarizerNode.updateUsedRoutes`, and that propagates to child summarizerNodes.

And remove `IFluidDataStoreChannel.updateUsedRoutes` altogether?

Reply:
There is a separation of responsibility between data store and summarizer node. Summarizer nodes only track and update things that are needed for incremental summaries. Data stores track everything else that's needed for regular opertaion.

The used routes are needed for 2 purposes - For keeping track of whether it changed between summaries for incremental summary. It allows data stores (and other nodes) to do their own sub-object clearn ups. For example, collection instances GC or sub-DDS objects GC. These will not be done by the runtime and the objects are in charge.

The case where data store is not loaded is slightly different because until data store is loaded, GC / summaries are incremental so it's fine for summarizer node to track them and update child nodes. After that, it's not up to the summarizer node because the objects owning the summarizer node has to be notified as well.